sui/token_bridge/sources/bridge_state.move

```diff
@@ -32,10 +30,40 @@ module token_bridge::bridge_state {
    friend token_bridge::transfer_tokens;
    #[test_only]
    friend token_bridge::test_bridge_state;
+   #[test_only]
+   friend token_bridge::token_bridge_vaa_test;
+
+   /// Wrapper around CoinType so that CoinType is "storable", for example in a dynamic map
+   struct CoinTypeWrapper<phantom CoinType> has copy, drop, store {}
 
-   /// TODO - The origin chain and address of a token.  In case of native tokens
-   ///        what do we set the token_address to? For Aptos it was the hash of the
-   ///        deployer + module_name + struct_name, for Sui might have to do differently
+   /// WrappedAssetInfo is a Sui object (has key ability) that functions as a Value
+   /// in a dynamic object mapping that maps Names => Values. It stores metadata about
+   /// a particular CoinType, or equivalently a CoinTypeWrapper.
+   struct WrappedAssetInfo has key, store {
+      id: UID,
+      token_chain: U16,
```

Comment:
why not put an `OriginInfo` in here? (same for `NativeAssetInfo`)

Reply:
I can change this later.